### ISSUE TITLE
RGB values are inverted when publishing from PTS in the potree viewer

### ISSUE THREAD

**Issue by umbe1987** [potree/PotreeConverter#181]
I have a PTS file in the form of:
nrows
x y z s r g b
When I convert it with PotreeConverter with the generate page option (-p), the RGB colors in the viewer seems to be inverted.
E.g. If the pixel has R:219, G:56, B:94, it should be close to purple, but it is bluish-green instead.
I managed to understand that the colors in the potree viewer are seen as if 0 is white and 255 is black. So in the example above I have a color that is represented as R:94, G:56, B:219 (255 minus the original value, i.e. as if RGB would be "BGR").
The image should have the colors of "example.jpg" (attached), but when I publish my pointcloud the colors are inverted in the Potree Viewer. The PTS I have is "test.pts" in the zip annex (I wasn't able to attach the zip file, so please remove the ".txt" extension from the filename to see it).
The example file I used is taken from http://gis.stackexchange.com/questions/138611/assigning-rgb-values-from-geotiff-image-to-lidar-data-using-r.
[test.zip.txt](https://github.com/potree/PotreeConverter/files/73365/test.zip.txt)
![example](https://cloud.githubusercontent.com/assets/15016826/12022095/e3f332f2-ad8c-11e5-9f65-3db076f98afa.JPG)

**Comment by bloomcreation** [comment]
I have the same issue, also on a PTS File.
The color are good in Autodesk RECAP, and after the convertion (with -p option) i get the same error.

**Comment by mojovski** [comment]
Having same issues here using pts files from Leica. Would be great to be able to load them in PotreeConverter without converting them.
Is there a way to set the positions of the color values in the file-row via cmd parameters?

**Comment by mradamcox** [comment]
I had the same color inversion issue. My column order was correct, but as in the OP, the color scale was backward (255 instead of 0, e.g.).

Here's the `awk` snippet I used on a linux remote machine to convert my original data to a version that displayed properly:
```
nohup awk '
  function abs(v) {return v < 0 ? -v : v}
  {print $1, $2, $3, $4, abs($5-255), abs($6-255), abs($7-255)}' MyPointcloud.pts > MyPointcloud_colorfix.pts &
```

Amazingly it only took about 20 minutes to convert a 20gb file, and then I ran the potree converter and the colors showed up correctly.
